File: sample-smart-contract-cosmos-wasm.go
package main

import (
	"fmt"

	cosmwasm "github.com/cosmos/cosmwasm/go/cosmwasm"
)

func main() {
	vm, err := cosmwasm.NewVM(1024)
	if err != nil {
		panic(err)
	}

	contractAddr, err := vm.DeployContract([]byte("contract.wasm"))
	if err != nil {
		panic(err)
	}

	result, err := vm.CallContract(contractAddr, []byte("hello"), nil)
	if err != nil {
		panic(err)
	}

	fmt.Println("Result:", result)
}

File: sample-smart-contract-cosmos-wasm.rs
extern crate cosmwasm_vm;

use cosmwasm_vm::{error::Error, runtime};

fn main() -> Result<(), Error> {
    let mut runtime = runtime::Builder::new()
        .with_memory(1024)
        .with_trait(Box::new(MyTraitImpl))
        .build()?;

    let wasm = include_bytes!("contract.wasm");
    let contract_address = runtime.deploy_contract(wasm)?;

    let result = runtime.call_contract(contract_address, b"hello", &[])?;
    println!("Result: {:?}", result);

    Ok(())
}
